src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using NBitcoin;
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.Networks;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests
+{
+    public class ConsensusManagerTests
+    {
+        private readonly Network network;
+
+        private const string walletAccount = "account 0";
+        private const string walletName = "mywallet";
+        private const string walletPassword = "123456";
+
+        public ConsensusManagerTests()
+        {
+            this.network = new StratisRegTest();
+            this.network.Consensus.Options = new ConsensusOptionsTest();
+
+            Type consensusType = typeof(NBitcoin.Consensus);
+            consensusType.GetProperty("MaxReorgLength").SetValue(this.network.Consensus, (uint)20);
+        }
+
+        private class ConsensusOptionsTest : PosConsensusOptions
+        {
+            public override int GetStakeMinConfirmations(int height, Network network)
+            {
+                return height < 10 ? 5 : 10;
```

Comment:
Why this condition?
Under 10 min confirmations is only 5, can you explain more?

Reply:
This was to simulate a block validation failure so that we can eventually disconnect it due to a `MaxReorgViolation`.  I'm investigating further.

Reply:
OK so this doesnt actually get triggered in this integration test as Im not actually working with stake blocks.

I'm going to create a second test to do that.